nssa/src/state.rs

```diff
@@ -2221,6 +2223,1547 @@ pub mod tests {
+                nonce: 0,
+            },
+            _ => panic!("Invalid selection"),
+        }
+    }
+
+    fn amm_state_constructor() -> V02State {
+        let initial_data = [];
+        let mut state =
+            V02State::new_with_genesis_accounts(&initial_data, &[]).with_test_programs();
+        state.force_insert_account(
+            helper_id_constructor(IdEnum::PoolDefinitionId),
+            helper_account_constructor(AccountsEnum::PoolDefinitionInit),
+        );
+        state.force_insert_account(
+            helper_id_constructor(IdEnum::TokenADefinitionId),
+            helper_account_constructor(AccountsEnum::TokenADefinitionAcc),
+        );
+        state.force_insert_account(
+            helper_id_constructor(IdEnum::TokenBDefinitionId),
+            helper_account_constructor(AccountsEnum::TokenBDefinitionAcc),
+        );
+        state.force_insert_account(
+            helper_id_constructor(IdEnum::TokenLPDefinitionId),
+            helper_account_constructor(AccountsEnum::TokenLPDefinitionAcc),
+        );
+        state.force_insert_account(
+            helper_id_constructor(IdEnum::UserTokenAId),
+            helper_account_constructor(AccountsEnum::UserTokenAHolding),
+        );
+        state.force_insert_account(
+            helper_id_constructor(IdEnum::UserTokenBId),
+            helper_account_constructor(AccountsEnum::UserTokenBHolding),
+        );
+        state.force_insert_account(
+            helper_id_constructor(IdEnum::UserTokenLPId),
+            helper_account_constructor(AccountsEnum::UserTokenLPHolding),
+        );
+        state.force_insert_account(
+            helper_id_constructor(IdEnum::VaultAId),
+            helper_account_constructor(AccountsEnum::VaultAInit),
+        );
+        state.force_insert_account(
+            helper_id_constructor(IdEnum::VaultBId),
+            helper_account_constructor(AccountsEnum::VaultBInit),
+        );
+
+        state
+    }
+
+    fn amm_state_constructor_for_new_def() -> V02State {
+        let initial_data = [];
+        let mut state =
+            V02State::new_with_genesis_accounts(&initial_data, &[]).with_test_programs();
+        state.force_insert_account(
+            helper_id_constructor(IdEnum::TokenADefinitionId),
+            helper_account_constructor(AccountsEnum::TokenADefinitionAcc),
+        );
+        state.force_insert_account(
+            helper_id_constructor(IdEnum::TokenBDefinitionId),
+            helper_account_constructor(AccountsEnum::TokenBDefinitionAcc),
+        );
+        state.force_insert_account(
+            helper_id_constructor(IdEnum::UserTokenAId),
+            helper_account_constructor(AccountsEnum::UserTokenAHolding),
+        );
+        state.force_insert_account(
+            helper_id_constructor(IdEnum::UserTokenBId),
+            helper_account_constructor(AccountsEnum::UserTokenBHolding),
+        );
+        state
+    }
+
+    #[test]
+    fn test_simple_amm_remove() {
```

Comment:
What's the reason to place tests here and not in `amm.rs`?

Reply:
It can't be inside program methods for that reason. But seems like `state.rs` is not the best place either. These are tests specific to the amm program and probably best to have them in a separate directory for the lssa std lib. But for now I would not move them. We can improve that later so the PR gets merged soon.